src/components/editor/filters/comment.js

```diff
@@ -14,15 +14,33 @@ function BlockCommentsControls( { clientId } ) {
 		getSelectedComment,
 		selectionStart,
 		selectionEnd,
-	} = useSelect(
-		( select ) => ( {
+		selectedText,
+	} = useSelect( ( select ) => {
+		const _selectionEnd = select( 'core/block-editor' ).getSelectionEnd();
+		const _selectionStart = select(
+			'core/block-editor'
+		).getSelectionStart();
+
+		let _selectedText = null;
+		const { offset: startOffset, attributeKey } = _selectionStart;
+		const { offset: endOffset } = _selectionEnd;
+
+		// we only support storing text now.
+		if ( attributeKey === 'content' ) {
+			const block = select( 'core/block-editor' ).getBlock( clientId );
+			_selectedText = block.attributes[ attributeKey ].slice(
+				startOffset,
+				endOffset
+			);
```

Comment:
I think the attributeKey for the block and the attributeKey for the RichText are not always the same thing, so this will break in some blocks. For instance in a Table block, there's multiple RichTexts but a single attribute.